File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/di/bean/BeanDefinition.java
package io.github.potjerodekool.openapi.internal.di.bean;

import java.lang.annotation.Annotation;
import java.lang.reflect.Method;

public interface BeanDefinition {

    Object getAutoConfigInstance();

    java.util.Map<Class<?>, Annotation> getAnnotations();

    Class<?> getBeanType();

    Method getBeanMethod();
}

File: code-generator/src/main/kotlin/io/github/potjerodekool/openapi/internal/di/bean/BeanDefinition.kt
package io.github.potjerodekool.openapi.internal.di.bean

import java.lang.reflect.Method
import kotlin.reflect.KClass

interface BeanDefinition {
    val autoConfigInstance: Any?
    val annotations: Map<KClass<*>, Annotation>
    val beanType: KClass<*>?
    val beanMethod: Method?
}
